packages/flutter_tools/lib/src/run_hot.dart

```diff
@@ -1270,6 +1270,15 @@ class ProjectFileInvalidator {
     if (updatedAt != null && updatedAt.isAfter(lastCompiled)) {
       invalidatedFiles.add(packageUri);
       packageConfig = await _createPackageConfig(packagesPath);
+      // The frontend_server might be monitoring the package_config.json file.
+      if (_fileSystem.path.basename(packagesPath) == '.packages') {
+        final File packageConfigFile = _fileSystem.file(packagesPath)
+          .parent.childDirectory('.dart_tool')
+          .childFile('package_config.json');
```

Comment:
Why not check directly by checking the modification date as is done with the .packages file?

Reply:
I could, but In theory they should only be modified together. This also becomes obsolete after https://github.com/flutter/flutter/issues/55249

Reply:
Ack. Consider adding a comment or two with TODOs pointing to the relevant issues.

Reply:
Done